### PR TITLE
feature: get token list from ks setting instead of ks-asset

### PR DESCRIPTION
**Pull request by TienPQ149**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -43,7 +43,12 @@
   "rules": {
     "unused-imports/no-unused-imports": "error",
     "@typescript-eslint/explicit-function-return-type": "off",
-    "prettier/prettier": "error",
+    "prettier/prettier": [
+      "error",
+      { 
+        "endOfLine":"auto" 
+      }
```
Comment (namgold) [inline .eslintrc.json:50]: Please revert it

File: src/utils/getTokenList.ts [hunk 2]
```diff
@@ -75,7 +75,15 @@ export default async function getTokenList(
     }
 
     const [json] = await Promise.all([response.json(), tokenListValidator])
-    if (BYPASS_LIST.indexOf(listUrl) >= 0) return json
+    const parsedData = !!json.data
```
Comment (namgold) [inline src/utils/getTokenList.ts:78]: Should be `json?.data?.tokens ? ...`

Or `json?.data?.tokens?.length ? ...`

Comment (TienPQ149) [reply src/utils/getTokenList.ts:78]: totally agree, should check carefully to prevent crash

**Review by namgold — APPROVED** [review]

File: src/utils/getTokenList.ts [hunk 3]
```diff
@@ -75,7 +75,15 @@ export default async function getTokenList(
...
-    if (BYPASS_LIST.indexOf(listUrl) >= 0) return json
+    const parsedData = !!json?.data?.tokens
+      ? {
+          tokens: json.data.tokens,
+          name: 'KyberSwap Token List',
+          logoURI: 'https://kyberswap.com/favicon.png',
+          keywords: ['kyberswap', 'dmmexchange'],
+        }
+      : json
+    if (BYPASS_LIST.indexOf(listUrl) >= 0) return parsedData
```
Comment (nguyenhuudungz) [inline src/utils/getTokenList.ts:86]: Why need this? It return `parsedData` below.

**Review by nguyenhuudungz — APPROVED** [review]

**Pull request merged**